apps/web/components/apps/AppPage.tsx

```diff
@@ -168,6 +173,91 @@ export const AppPage = ({
     }
   }, []);
 
+  const installOrDisconnectAppButton = () => {
+    if (appDbQuery.isPending) {
+      return <SkeletonButton className="h-10 w-24" />;
+    }
+
+    return (
+      <div className="flex items-center space-x-3">
+        {isGlobal ||
+          (existingCredentials.length > 0 && allowedMultipleInstalls ? (
+            <div className="flex space-x-3">
+              <Button StartIcon="check" color="secondary" disabled>
+                {existingCredentials.length > 0
+                  ? t("active_install", { count: existingCredentials.length })
+                  : t("default")}
+              </Button>
+              {!isGlobal && !appInstalled && (
+                <InstallAppButton
+                  type={type}
+                  disableInstall={disableInstall}
+                  teamsPlanRequired={teamsPlanRequired}
+                  render={({ useDefaultComponent, ...props }) => {
+                    if (useDefaultComponent) {
+                      props = {
+                        ...props,
+                        onClick: () => {
+                          handleAppInstall();
+                        },
+                        loading: isLoading,
+                      };
+                    }
+                    return <InstallAppButtonChild multiInstall paid={paid} {...props} />;
+                  }}
+                />
+              )}
+            </div>
+          ) : (
+            !appInstalled && (
+              <InstallAppButton
+                type={type}
+                disableInstall={disableInstall}
+                teamsPlanRequired={teamsPlanRequired}
+                render={({ useDefaultComponent, ...props }) => {
+                  if (useDefaultComponent) {
+                    props = {
+                      ...props,
+                      onClick: () => {
+                        handleAppInstall();
+                      },
+                      loading: isLoading,
+                    };
+                  }
+                  return (
+                    <InstallAppButtonChild
+                      credentials={appDbQuery.data?.credentials}
+                      paid={paid}
+                      {...props}
+                    />
+                  );
+                }}
+              />
+            )
+          ))}
+
+        {existingCredentials.length > 0 && (
+          <>
+            {existingCredentials.length > 1 ? (
+              <MultiDisconnectIntegration
+                credentials={existingCredentials}
+                onSuccess={() => appDbQuery.refetch()}
+              />
+            ) : (
+              <DisconnectIntegration
+                buttonProps={{ color: "secondary" }}
+                label={t("disconnect")}
+                credentialId={Number(existingCredentials[0].id)}
+                teamId={existingCredentials[0].teamId}
+                onSuccess={() => appDbQuery.refetch()}
+              />
```

Comment:
I think we could make `MultiDisconnectIntegration` a wrapper over `DisconnectIntegration` so that it can handle single as well multiple disconnects. It keeps the consumer logic simple